src/anet.c

```diff
@@ -381,16 +381,16 @@ static int anetSetReuseAddr(char *err, int fd) {
     return ANET_OK;
 }
 
-static int anetCreateSocket(char *err, int domain) {
+static int anetCreateSocket(char *err, int domain, int type, int protocol, int sockopts) {
     int s;
-    if ((s = socket(domain, SOCK_STREAM, 0)) == -1) {
+    if ((s = socket(domain, type, protocol)) == ANET_ERR) {
```

Comment:
i think ANET_ERRO is meant to be used by Redis's internal functions, for the system call we better keep checking `-1`

Reply:
Consider it done.